book/src/custom.tex

```diff
@@ -54,134 +56,315 @@ \subsection{New Commands} \label{sec:new_commands}
 \begin{itemize}
 \item \txsit{not so}{short}
 \item \txsit{very}{long}
 \end{itemize}
 \end{example}
 
+If your command accepts an optional argument but the user does not supply one,
+a special marker \cargv{-NoValue-} will be inserted instead.
+\begin{example}
+\NewDocumentCommand{\txsit}{om}
+{This is the \emph{#1}
+  #2 Introduction to \LaTeXe}
+
+% in the document body:
+\begin{itemize}
+\item \txsit{definitive}
+\item \txsit[very]{long}
+\end{itemize}
+\end{example}
+In order to test whether the user supplied a value, use
+\begin{lscommand}
+  \ci*{IfValueTF}[argument:m ! value version:m ! no value version:m] 
+\end{lscommand}
+macro.
+\begin{example}
+\NewDocumentCommand{\MyCommand}{o}{%
+  \IfValueTF {#1} {%
+    Optional argument: #1.%
+  } {
+    No optional argument given.%
+  }%
+}
+
+\MyCommand\\
+\MyCommand[hello]
+\end{example}
+
+There are two variations of it: \ci{IfValueT} and \ci{IfValueF} which may
+be used if you only need output for one of the branches. Thus the example with
+\cargv{-NoValue-} in the output could be fixed by writing
+\begin{example}
+\NewDocumentCommand{\txsit}{om}
+{This is the
+  \IfValueT{#1}{\emph{#1} }%
+  #2 Introduction to \LaTeXe}
+
+% in the document body:
+\begin{itemize}
+\item \txsit{definitive}
+\item \txsit[very]{long}
+\end{itemize}
+\end{example}
+There are also commands \ci{IfNoValueTF}, \ci{IfNoValueT} and \ci{IfNoValueF}
+which work exactly the same, but the value\slash no-value branches are swapped.
+
+Often you will want to use optional arguments when present but use some default
+values when the user does not provide them. This could be achieved by
+\ci{IfValueTF}, but a much better approach exists: \cargv{O} argument
+specification. It works like \cargv{o} but allows us to specify default value
+if no value is supplied. Thus we can write
+
+\begin{example}
+\NewDocumentCommand{\txsit}{O{not so}m}
+{This is the \emph{#1}
+  #2 Introduction to \LaTeXe}
+
+% in the document body:
+\begin{itemize}
+\item \txsit{definitive}
+\item \txsit[very]{long}
+\end{itemize}
+\end{example}
+
+Another useful argument specification is \cargv{s}, short for star. This
+argument enables us to provide different definitions based on whether the
+starred or non-starred version of command was issued by the user. It uses
+\ci{IfBooleanTF} command (and its variations \cih{IfBooleanT} \cih{IfBooleanF})
+that works analogously to the \ci{IfValueTF} command. 
+
+\begin{example}
+\NewDocumentCommand{\txsit}{sO{not so}m}
+{This is the \emph{#2}
+  #3 Introduction to \LaTeXe%
+  \IfBooleanT{#1}{%
+    : Superstar Edition%
+  }%
+}
+
+% in the document body:
+\begin{itemize}
+\item \txsit{long}
+\item \txsit*{long}
+\end{itemize}
+\end{example}
+
+These are just the most common argument specifications. For a full overview of
+those take a look at the documentation of \pai*{xparse} package.\footnote{This
+package is included by default in new \LaTeX{} versions, so you probably do not
+need to include it manually.}
```

Comment:
```suggestion
package is part of the current \LaTeX{} release, so you probably do not
need to install it manually.}
```

Reply:
This comment wasn't about installing but about doing `\usepackage{xparse}`, hence the word include. Do you think other word would be more appropriate?

Reply:
oops :)

Reply:
how about this wording ?